src/processing.jl

```diff
@@ -7,16 +7,23 @@ Preprocessing functions
 =#
 
 """
-    resample(d::RheoTimeData [, t = array of time value or range, dt = time step required, scale = multiplicator to apply to existing sampling rate] )
+    resample(d::RheoTimeData [, t, dt, scale )
 
-Resample data.
+Resample the data using 1D spline extrapolation (using the Dierckx.jl package).
 
-Usage:
+# Arguments
+
+- `d`: data as a `RheoTimeData` struct containing time with stress and/or strain. Without additional parameters, the data is resampled to provide a uniform sampling at constant number of timepoints.
+- `t`: an array or range that determines the timepoints where the data will be provided.
+- `dt`: if the array `t` is not provided, the parameter `dt` will set the timestep for a uniform resampling of the data.
+- `scale`: instead of specifying particular time points or timestep, an overall multiplicator on the sampling rate can be provided. This could down-sample (`scale`<1) or upsample (`scale`>1). If timesteps are non uniform, it would interpolate values accordingly.
+
+# Examples:
 
-* 'resample(d)' would keep the number of sampling points the same but interpolate to set a uniform time step.
-* 'resample(d, t=-1:0.1:10)' would resample by interpolation to generate a new dataset with time points given by the range or array values passed with keyword 't'.
-* 'resample(d, dt=0.1)' would resample by interpolation to generate a new dataset with time step 'dt'.
-* 'resample(d, scale=2)' would resample by multiplying the sampling rate by 'scale'. This could down-sample (scale<1) or upsample (scale>1). If timesteps are non uniform, it would interpolate values accordingly.
+* `resample(d)` keeps the number of sampling points the same but interpolates to set a uniform time step.
```

Comment:
Tough Cookie Warning :cookie: :smile: 

Looks better to me, but note that example blocks normally contain some copy-pastable Julia code that is completely self contained (i.e. `d` would be defined). See for example

https://docs.julialang.org/en/v1/base/c/#Base.@cfunction
https://docs.julialang.org/en/v1/base/math/#Base.max

You're wording after the usage examples is very clear and helpful. If you don't want to bother with fully fledged examples maybe the usage wording could be integrated with the arguments and that would be clear enough, then the examples block could be removed.

Just something to bear in mind if we are being pedantic, probably best to be consistent with using `-` for bulletpoints rather than a mix of `-` and `*`.

Anyway, I'll leave this in your responsible hands now :+1:

Reply:
Good point about the self-contained examples. I fear this would make the presentation very heavy though if we want to show different cases, including outputs, etc. I added a definition for the variable d though, as this was missing. Will keep that in mind in the future.

OK with consistency between - and * for lists. Somehow they seem to be rendered in the same manner.

Reply:
Yup - and * are rendered the same by HTML markdown renderer, my comment was for those viewing docstrings directly in the source code.

I'm not sure there's anything wrong with heavy docs, but as I mentioned, the way you've written the examples, the wording of each one nicely slots in the description of each argument so an examples section may not be essential at all here. 

Separately, for examples of heavier docstring examples, a quick scan led me to finding these examples:

https://docs.julialang.org/en/v1/base/c/#Core.Ref

https://docs.julialang.org/en/v1/base/c/#Base.copyto!

It's quite nice in my humble opinion, especially for more complicated functionality.

Reply:
Also, see point 6:

https://docs.julialang.org/en/v1/manual/documentation/

If docstring examples are self-contained, we can actually start using them as additional tests, to make sure they are up to date.